docs/user-guide/maps/index.rst

```diff
@@ -264,11 +265,26 @@ A `MapCoord` with named axes can be created by calling `MapCoord.create` with a
     [ 100 1000] [ 100 1000]
 
 
-Spatial axes must be named ``lon`` and ``lat``. `MapCoord` objects created with
+Spatial axes must be named ``lon`` and ``lat``. `~gammapy.maps.MapCoord` objects created with
 named axes do not need to have the same axis ordering as the map geometry.
 However the name of the axis must match the name of the corresponding map
 geometry axis.
 
+Note: it is possible to have an arbitrary number of additional non-spatial dimensions,
+however, the user must ensure that the array has been correctly broadcasted. Here, we
+demonstrate this using an energy array with length greater than 2.
```

Comment:
sorry, I thought the energy axis was reshaped into a 2d array

Reply:
can we do the broadcasting internally, I this this can be very confusing for users, especially as the shape differently if we do :
```
print(c)
MapCoord

	axes     : ['lon', 'lat', 'energy']
	shape    : (2, 3)
	ndim     : 3
	frame : icrs 
```

Reply:
> I think this can be very confusing for users, especially as the shape is displayed differently if we do :

Yeah this is an old issue #1180 that we decided not to address before 1.0 and I am not sure we want to do it now!

Reply:
> Is it a realistic use case to use " an energy array with length greater than 2" ? should we rather show a map with more than one additional non-spatial dimensions instead ?

Do you mean to show a map with more than just energy as the non-spatial dimension?

Reply:
> Do you mean to show a map with more than just energy as the non-spatial dimension?

yes maybe time or energy_true in addition

Reply:
```
energy = np.array([1000, 3000, 5000]).reshape(-1, 1, 1)
energy_true = np.arange(500, 6000, 500).reshape(1, -1, 1)

#maybe using np.newaxis is more clear
energy = np.array([1000, 3000, 5000])[:, np.newaxis, np.newaxis]
energy_true = np.arange(500, 6000, 500)[np.newaxis, : ,np.newaxis]

c = MapCoord.create(dict(skycoord=position,energy=energy,energy_true=energy_true))
```